tobler/util/util.py

```diff
@@ -116,15 +113,36 @@ def h3fy(source, resolution=6, clip=False, buffer=False, return_geoms=True):
     clipper = source
 
     if not source.crs.is_geographic:
+        crs_units = source.crs.to_dict()["units"]
         if buffer:
+            if not crs_units in ["m", "us-ft"]:
+                raise ValueError(
+                    f"The coordinate system uses an unknown measurement unit: {crs_units} ."
+                    "Currently, acceptably inputs include meters or U.S. feet. Please reproject the input "
+                    "geodataframe to a system that uses these units or pass `buffer=False`"
+                )
             clipper = source.to_crs(4326)
             distance = circumradius(resolution)
+            if crs_units == "ft-us":
+                distance = distance * 3.281
             source = source.buffer(distance).to_crs(4326)
         else:
             source = source.to_crs(4326)
 
-    source_unary = source.unary_union
+    # if CRS is geographic but user wants a buffer, we need to estimate
+    if buffer:
```

Comment:
This probably needs to be behind `else` from `if not source.crs.is_geographic:`. Otherwise it is probably okay.